lib/clarkson-core-objects.php

```diff
@@ -135,12 +100,7 @@ public function get_objects( $posts ) {
 	 *
 	 * @return \Clarkson_Object Clarkson Post object.
 	 */
-	public function get_object( $post ) {
-		if ( ! $post instanceof WP_Post && is_int( (int) $post ) ) {
-			_doing_it_wrong( __METHOD__, 'Deprecated calling of get_object with an ID. Use a `WP_Post` instead', '1.0.0' );
-			$post = get_post( $post );
-		}
-
+	public function get_object( \WP_Post $post ) {
```

Comment:
Deze mogen dus wel een return type krijgen.

Reply:
Wat moet hier de return type zijn? Clarkson_Object, omdat een overschrijvend object altijd Clarkson_Object moet extenden?

Reply:
Klopt. En als de implementerende functie in een thema iets specifiekers nodig heeft moet daar (in het thema) een instanceof check gedaan worden.